packages/ozone/src/mod-service/lang.ts

```diff
@@ -40,6 +45,27 @@ export class ModerationLangService {
     }
   }
 
+  getTextFromRecord(
+    uri: string,
+    recordValue?: Record<string, unknown>,
+  ): string | undefined {
+    const atUri = new AtUri(uri)
+    let text: string | undefined
+    if (
+      [ids.AppBskyGraphList, ids.AppBskyFeedGenerator].includes(
+        atUri.collection,
+      )
+    ) {
+      text = (recordValue?.description || recordValue?.name) as string
+    }
```

Comment:
Another way to slice this is to use `AppBskyGraphList.isRecord(recordValue)` to check the record's `$type` (i.e. its collection). A nice upside is that it ends-up attributing `recordValue` the proper typescript type (assuming `recordValue` is a valid record), which could help avoid the need to cast.  E.g. might look something like:
```ts
if (AppBskyGraphList.isRecord(recordValue)) {
  text = recordValue.description || recordValue.name
} else if (AppBskyFeedGenerator.isRecord(recordValue)) {
  text = recordValue.description || recordValue.name
} else if (AppBskyFeedPost.isRecord(recordValue)) {
  text = recordValue.text
}
```

Reply:
much cleaner! thanks